src/api/publish.rs

```diff
@@ -4,9 +4,16 @@ use uuid::Uuid;
 /// The publish request.
 #[derive(Deserialize, Debug)]
 pub struct PublishRequest {
-    pub name: String,
-    pub version: String,
-    pub upload_id: String,
+    pub package_name: String,
+    pub upload_id: Uuid,
+    pub num: String,
+    pub package_description: Option<String>,
+    pub repository: Option<String>,
```

Comment:
nit: could we use the `url` crate here instead of `String` to urls are validated before being stored? 

```rust
use url::Url;

pub repository: Option<Url>,
pub documentation: Option<Url>, 
pub homepage: Option<Url>,
pub urls: Vec<Option<Url>>,
```

Reply:
Good catch. I had input validation on my todo list and totally forgot to do it. We need it for package name as well, and possibly a few other fields.